### PR TITLE
Tag Files

### PR DESCRIPTION
**Pull request by pascaldekloe**
Allows Java annotations and Go field tags.

### CODE REVIEW

File: colfer.go [hunk 1]
```diff
@@ -147,6 +151,8 @@ type Struct struct {
 	Fields []*Field
 	// SchemaFile is the source filename.
 	SchemaFile string
+	// TagAdd is an optional source code addition.
+	TagAdd string
```
Comment (guilt) [inline colfer.go:155]: If I wanted multiple tags on different lines, such as on Java/Python, wouldn't it be easier to make this a: TagAdd[] string instead?

**Comment by guilt** [comment]
Thank you for doing this. 👍 This solves my current problem, but was asking about the other part, where if we need to assign multiple tags?

File: colfer.go (hunk 1 above)
Comment (pascaldekloe) [reply colfer.go:155]: Sure. It requires a bit more thought on the input format though.

Why would you want multiple lines?

**Comment by pascaldekloe** [comment]
Mutiple Java annotations can follow on the same line right? Dito for Go and Python.

File: colfer.go (hunk 1 above)
Comment (guilt) [reply colfer.go:155]: Java annotations can become quite unwieldy, especially if qualifying them with package names too:

```
public class Test {
	@Deprecated @SuppressWarnings({"unchecked", "deprecation"}) @com.fasterxml.jackson.annotation.JsonProperty("x") @com.google.gson.annotations.SerializedName(value = "x", alternate = "X")
	String x;
}
```

**Comment by guilt** [comment]
Yeah, I'll approve this PR for now, given that we are only looking at readability, for now. :-)

**Comment by pascaldekloe** [comment]
Let's not rush such feature. I'm still not sure this is right.

Shoud the compiler enforce tagging each field (possibly with an empty string) to prevent mistakes? Or maybe log something (on verbose level)?
Are multiple declarations for the same field OK?
Should we allow the line to continue (with a tailing '\')?
Etc..

**Comment by guilt** [comment]
Here's answers to some of these questions, from what I can gather: 

- We should not need to tag fields unless necessary. 
- Multiple declarations for the same field should be OK. I can't think of anything apart from `@Verbose @Verbose` etc. where duplicate tags are allowed. So multiple lines, but unique for every field.
- Allowing the line to continue is a bad idea, and most certainly will be the cause of some programming bug
  where a trailing unseen character is missed when deletion.

**Comment by pascaldekloe** [comment]
Good point on the leading slash mistake @guilt. One advantage is that you have everything nicely together in one overview. E.g., you can clearly see the remaining after deleting a line. On the other hand, a `colf -t json.tags,validation.tags …` could be nice too. I do not like the idea of automatic deduplication. Image deleting a line while the generated code stays the same. Could throw an error on duplicates though. Not sure if it's worth the effort.

**Comment by guilt** [comment]
Agreed on the duplicate handling. It was a point of caution. Yeah, I think those may be the ones we could get together in this, and get it in.

**Comment by pascaldekloe** [comment]
Still unsure about the multiple line declaration. This is more flexible (separtion of concerns possible) yet I do like to keep the declarations somewhat together. Line continuation with leading whitespace (like Python) would keep things simple.

```
test.req.ID     @javax.validation.constraints.Min(1)
        @javax.validation.constraints.Max(0xffffffffL)
test.req.code   @javax.validation.constraints.NotNull
        @javax.validation.constraints.Size(max=8)
```

**Comment by guilt** [comment]
These are smaller lines with constraints, but the ones I've seen are usually much longer, deal with a variety of things pattern, null ability, length, JSON field names, XML tag names... Bunch of them. I know that this kind of bloat often comes with usage, but this well may be the right option for Java.

**Comment by pascaldekloe** [comment]
Java enterprise ™️ devs be like:

```
esb.AbstractInfo2Request @org.apache.frameworky.v3.annotations.DecorationFilter(
esb.AbstractInfo2Request     factory=com.example.divisionx.dao.util.CommonHelperFilterBuilder.class
esb.AbstractInfo2Request     factory-config= [
…
esb.AbstractInfo2Request )
```

😆 

The setup looks acceptable now. Problems don't get any better by adding more structure here. I'll apply this one with a few more detail in favour of #64, if that's OK with you @guilt?

**Comment by guilt** [comment]
This is great, thank you so much!

**Comment by guilt** [comment]
This is great, thank you so much!

**Comment by pascaldekloe** [comment]
We did this together. Cheers!

**Pull request merged**